x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_status/monitor_status_panel.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { throttle } from 'lodash';
+import { EuiPanel, useEuiTheme, EuiResizeObserver } from '@elastic/eui';
+import { Chart, Settings, Heatmap, ScaleType } from '@elastic/charts';
+
+import { scheduleToMinutes } from '../../../../../../common/lib/schedule_to_time';
+import { useSyntheticsRefreshContext } from '../../../contexts/synthetics_refresh_context';
+
+import { useSelectedMonitor } from '../hooks/use_selected_monitor';
+import { usePingStatuses } from '../hooks/use_ping_statuses';
+import { MonitorStatusHeader } from './monitor_status_header';
+import { MonitorStatusCellTooltip } from './monitor_status_cell_tooltip';
+import { MonitorStatusLegend } from './monitor_status_legend';
+import { getMonitorStatusChartTheme } from './monitor_status_chart_theme';
+import {
+  getXAxisLabelFormatter,
+  dateToMilli,
+  createTimeBuckets,
+  createStatusTimeBins,
+  getColorBands,
+  CHART_CELL_WIDTH,
+  indexBinsByEndTime,
+  getBrushData,
+  MonitorStatusPanelProps,
+} from './monitor_status_data';
+
+export const MonitorStatusPanel = ({
+  from = 'now-24h',
+  to = 'now',
+  brushable = true,
+  periodCaption = undefined,
+  showViewHistoryButton = false,
+  onBrushed,
+}: MonitorStatusPanelProps) => {
+  const { euiTheme, colorMode } = useEuiTheme();
+  const { lastRefresh } = useSyntheticsRefreshContext();
+  const { monitor } = useSelectedMonitor();
+  const monitorInterval = Math.max(3, monitor?.schedule ? scheduleToMinutes(monitor?.schedule) : 3);
+
+  const fromMillis = dateToMilli(from);
+  const toMillis = dateToMilli(to);
+  const totalMinutes = Math.ceil(toMillis - fromMillis) / (1000 * 60);
+  const pingStatuses = usePingStatuses({
+    from: fromMillis,
+    to: toMillis,
+    size: Math.min(10000, Math.ceil((totalMinutes / monitorInterval) * 2)), // Acts as max size between from - to
+    monitorInterval,
+    lastRefresh,
+  });
+
+  const [binsAvailableByWidth, setBinsAvailableByWidth] = useState(50);
+  const intervalByWidth = Math.floor(
+    Math.max(monitorInterval, totalMinutes / binsAvailableByWidth)
+  );
+
+  // Disabling deps warning as we wanna throttle the callback
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const resizeHandler = useCallback(
+    throttle((e: { width: number; height: number }) => {
+      setBinsAvailableByWidth(Math.floor(e.width / CHART_CELL_WIDTH));
+    }, 500),
+    []
+  );
+
+  const { timeBins, timeBinsByEndTime, xDomain } = useMemo(() => {
+    const timeBuckets = createTimeBuckets(intervalByWidth, fromMillis, toMillis);
+    const bins = createStatusTimeBins(timeBuckets, pingStatuses);
+    const indexedBins = indexBinsByEndTime(bins);
+
+    const timeDomain = {
+      min: bins?.[0]?.end ?? fromMillis,
+      max: bins?.[bins.length - 1]?.end ?? toMillis,
+    };
+
+    return { timeBins: bins, timeBinsByEndTime: indexedBins, xDomain: timeDomain };
+  }, [intervalByWidth, pingStatuses, fromMillis, toMillis]);
```

Comment:
Does it make sense to extract this into a hook? to make component more stateless.